sdks/uniswapx-sdk/src/utils/dutchBlockDecay.ts

```diff
@@ -0,0 +1,120 @@
+import { BigNumber } from "ethers";
+
+import { V3Decay } from "../order";
+/*
+These functions mimic the smart contract functions as closely as possible to ensure that the same results are produced.
+Essentially Solidity translated to TypeScript.
+*/
+/* eslint-disable */
```

Comment:
nit: why disable? a bit hard to read

Reply:
The linter wants `next` and `prev` inside of `locateArrayPosition` to be declared as `const` even though they are both reassigned in the while-loop. While I could rewrite the function to an equivalent implementation, I intentionally want to preserve identical code to the contract so that there aren't any edge-case differences in math. 

Agreed that disabling linter for the whole file was too far. Removed and only included directives for the 2 lines above `next` and `prev`!

Reply:
No longer a concern after the new implementation of this functionality in the contract!